builder/builder-module.c

```diff
@@ -1520,18 +1550,43 @@ builder_module_build (BuilderModule  *self,
 
   /* Build and install */
 
-  if (meson || cmake_ninja)
-    make_cmd = "ninja";
+  if (simple)
+    {
+      guint length = g_strv_length (self->build_commands);
+      gint argc;
+      gchar **argv;
+      g_print ("Building with simple builsystem: %d build commands\n", length);
+
+      for (guint i = 0; i < length; i++)
+        {
+	  g_print ("%s\n", self->build_commands[i]);
+
+          if (!g_shell_parse_argv (self->build_commands[i], &argc, &argv, error))
+            return FALSE;
+
+
+          if (!build (app_dir, self->name, context, source_dir, build_dir_relative, build_args, env, error,
+                      argv[0], strv_arg, argv + 1, NULL))
+            return FALSE;
```

Comment:
These returns leak `argv`. You can do `g_autoptr(GStrv) args` and let it do its magic within this scope.